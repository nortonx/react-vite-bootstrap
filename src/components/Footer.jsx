import React from "react";
import { Link } from "react-router-dom";
const Footer = () => {

  return(
    <div className="container footer">
      <footer className="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
        <div className="col-md-4 d-flex align-items-center">
          <a href="/" className="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">Logo</a>
          <span className="mb-3 mb-md-0 text-muted">{new Date().getFullYear()} &copy;</span>
        </div>
        <ul className="nav col-md-4 justify-content-end list-unstyled d-flex">
          <li className="ms-3">
          <Link to="/" className="text-muted">Home</Link>
          </li>
          <li className="ms-3">
            <Link to="/about" className="text-muted">About</Link>
          </li>
        </ul>
      </footer>
    </div>
  )
}

export default Footer;