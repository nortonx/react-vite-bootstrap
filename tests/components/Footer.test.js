import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import Footer from "../../src/components/Footer"

describe("Footer", () => {
  it("should render the component and match snapshot", async () => {
    render(<Footer />, {wrapper: BrowserRouter})
    const element = await screen.findByTestId("footer");
    expect(element).toBeInTheDocument();
    expect(element).toMatchSnapshot();
  })
})