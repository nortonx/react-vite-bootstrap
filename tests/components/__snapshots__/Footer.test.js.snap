// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render the component and match snapshot 1`] = `
<div
  class="container footer"
  data-testid="footer"
>
  <footer
    class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top"
  >
    <div
      class="col-md-4 d-flex align-items-center"
    >
      <a
        class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1"
        href="/"
      >
        Logo
      </a>
      <span
        class="mb-3 mb-md-0 text-muted"
      >
        2023
         Â©
      </span>
    </div>
    <ul
      class="nav col-md-4 justify-content-end list-unstyled d-flex"
    >
      <li
        class="ms-3"
      >
        <a
          class="text-muted"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="ms-3"
      >
        <a
          class="text-muted"
          href="/form"
        >
          Form
        </a>
      </li>
      <li
        class="ms-3"
      >
        <a
          class="text-muted"
          href="/about"
        >
          About
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
